File: src/irc/yoni-commands.ts
import { EventEmitter } from "events";
import { IrcTags } from ".";

export enum YoniMessage {
    TurnOn = 1,
    TurnOff = 2,
}

const toStringMap = new Map<YoniMessage, string>([
    [YoniMessage.TurnOn, "Turning on"],
    [YoniMessage.TurnOff, "Turning off"]
]);

export type MessageFromYoni = {
    type: YoniMessage,
    displayName: string,
}

export function toStringMessageFromYoni(message: MessageFromYoni): string {
    return toStringMap.get(message.type);
}

const msgToEmit: {[key: string]: YoniMessage} = {
    "!enable": YoniMessage.TurnOn,
    "!disable": YoniMessage.TurnOff,
}

//@ts-ignore
export default function Commands(emitter: EventEmitter, tags: IrcTags, message: string): void {
    const type = msgToEmit[message];

    if (type) {
        emitter.emit("from-yoni", {
            type, displayName: tags["display-name"]
        });
    }
}

File: pkg/irc/yoni-commands.go
package irc

import (
	"strings"
)

func YoniCommands(Msg IrcMessage, channel chan IrcMessage) {
    if Msg.Name != "yonikosiner" {
        return
    }

    m := map[Key]MessageType{}
    m[Key{"!enable"}] = TurnOn
    m[Key{"!disable"}] = TurnOff

    yoniMessage := m[Key{strings.Fields(Msg.Message)[0]}]
    Msg.Type = MessageType(yoniMessage)

    go func() {
        if Msg.Type == TurnOn || Msg.Type == TurnOff {
            channel <- Msg
        }
    }()
}

